perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -273,6 +273,29 @@ public void deleteMembersGroup(PerunSession sess, Vo vo) throws InternalErrorExc
 	private void deleteAnyGroup(PerunSession sess, Group group, boolean forceDelete) throws InternalErrorException, RelationExistsException, GroupAlreadyRemovedException, GroupAlreadyRemovedFromResourceException, GroupNotExistsException, GroupRelationDoesNotExist, GroupRelationCannotBeRemoved {
 		Vo vo = this.getVo(sess, group);
 
+		//remove admins of this group
+		List<Group> adminGroups = getGroupsManagerImpl().getGroupAdmins(sess, group);
+
+		for (Group adminGroup : adminGroups) {
+			try {
+				AuthzResolverBlImpl.unsetRole(sess, adminGroup, group, Role.GROUPADMIN);
+			} catch (GroupNotAdminException e) {
+				log.warn("When trying to unsetRole GroupAdmin for group with id: " + adminGroup.getId() + ", in the group with id: " + group.getId() + " , this exception was thrown: " + e.toString());
+				//skip and log as warning
+			}
+		}
+
+		List<User> adminUsers = getGroupsManagerImpl().getAdmins(sess, group);
+
+		for (User adminUser : adminUsers) {
+			try {
+				AuthzResolverBlImpl.unsetRole(sess, adminUser, group, Role.GROUPADMIN);
+			} catch (UserNotAdminException e) {
+				log.warn("When trying to unsetRole GroupAdmin for user with id: " + adminUser.getId() + ", in the group with id: " + group.getId() + " , this exception was thrown: " + e.toString());
+				//skip and log as warning
+			}
+		}
```

Comment:
You should probably do this for all subgroups of this group.

Reply:
I just wonder, why we don't simply call ``deleteAnyGroup()`` on each subgroup, after we retrieve them and sort them in reverse order. By the quick look,  only difference is, that we don't care about their relations and delete them anyway (like we would call this main method with ``forceDelete=TRUE``).

Reply:
Also we might want to do all of this after we check members and subgroups, since if deletion is not forced, it will fail more quickly and has to rollback fewer changes.